db/controller/articlesId.controller.js

```diff
@@ -0,0 +1,19 @@
+const { getArticleById } = require("../models/articlesId.model");
+
+// Controller for fetching an article by ID
+exports.getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+  getArticleById(article_id)
+    .then((article) => {
+      if (!article) {
```

Comment:
Remember that .then() is for handling a promises resolving - i.e. we got the data everything is successful - and .catch is to handle errors - theres a problem and the promise has rejected - therefore we should make sure the model resolves or rejects so the controller can send the appropriate response in the appropriate place!
Refer back to the notes on error handling too to see a more appropriate way of creating this error